cmd/ipfs/main.go

```diff
@@ -473,6 +481,9 @@ func (i *cmdInvocation) setupInterruptHandler() {
 	sig := allInterruptSignals()
 
 	go func() {
+		// wait till the context is ready to be closed
+		<-ctx.ContextIsReadyToBeClosed
```

Comment:
We can't do `n.Close()` because it will cause the race conditions. We have to make sure the context is not going to be modified by `daemonFunc()` before we use any part of the context in `setupInterruptHandler()`.  And `n` is the part of the context.

Is it okay to `os.Exit()` during daemon startup?